src/resultsdisassemblypage.cpp

```diff
@@ -274,19 +274,58 @@ ResultsDisassemblyPage::ResultsDisassemblyPage(CostContextMenu* costContextMenu,
     connect(ui->assemblyView, &QTreeView::entered, this, updateFromDisassembly);
     connect(ui->sourceCodeView, &QTreeView::entered, this, updateFromSource);
 
-    ui->sourceCodeView->setContextMenuPolicy(Qt::CustomContextMenu);
-    connect(ui->sourceCodeView, &QTreeView::customContextMenuRequested, this, [this](QPoint point) {
-        const auto index = ui->sourceCodeView->indexAt(point);
-        const auto fileLine = index.data(SourceCodeModel::FileLineRole).value<Data::FileLine>();
-        if (!fileLine.isValid())
-            return;
+    auto createContextMenu = [](QTreeView* view, auto* model, int role, auto&& addEntries) {
+        auto gotoMenuWidget = new QWidget(view);
+        auto layout = new QHBoxLayout(gotoMenuWidget);
+        layout->setContentsMargins(0, 0, 0, 0);
+        auto label = new QLabel(QCoreApplication::translate("Util", "Goto: "), gotoMenuWidget);
+        layout->addWidget(label);
+        auto edit = new QLineEdit(gotoMenuWidget);
+        layout->addWidget(edit);
+        auto gotoAction = new QWidgetAction(view);
+        gotoAction->setDefaultWidget(gotoMenuWidget);
+
+        auto gotoLine = [view, model, role, edit] {
+            const auto lineNumber = edit->text().toInt();
+
+            // model->index()->data is somehow not possible
```

Comment:
what does this mean?

Reply:
If I try `model->index(..)->data()` the compiler throws an error

Reply:
same if I replace `QModelIndex` with auto.
Error message:
```
hotspot/src/resultsdisassemblypage.cpp:293:46: error: use 'template' keyword to treat 'value' as a
 dependent template name
            const auto offset = index.data().value<int>();
                                             ^
                                             template 
1 error generated.
```